### ISSUE TITLE
Scopes not resolved on custom builder instances (Post::query()->featured())

### ISSUE THREAD

**Issue by alies-dev** [psalm/psalm-plugin-laravel#630]
## Problem

When a model uses a custom query builder, scopes called on the builder instance are not resolved:

```php
// Works — goes through ModelMethodHandler::getReturnTypeForForwardedMethod
Post::featured(); // ✅ PostBuilder<Post>

// Fails — BuilderScopeHandler is only registered for Builder, not custom subclasses
Post::query()->featured(); // ❌ UndefinedMagicMethod on PostBuilder::featured
```

## Root cause

`BuilderScopeHandler` is registered for `Illuminate\Database\Eloquent\Builder` via `getClassLikeNames()`. When `Post::query()` returns `PostBuilder<Post>`, Psalm looks for `featured()` on `PostBuilder` — but the scope handler isn't registered for `PostBuilder`.

## Possible fix

In `ModelRegistrationHandler::registerHandlersForModel()`, when a custom builder is detected, also register `BuilderScopeHandler` for the custom builder class.

## Context

Discovered during PR #621 which added custom builder detection. The static call path works because it goes through `ModelMethodHandler` which handles scopes directly.
